content/graphql/guards-interceptors.md

```diff
@@ -47,21 +47,34 @@ The GraphQL context object returned by `GqlExecutionContext.create()` exposes a
 
 #### Exception filters
 
-Nest standard [exception filters](/exception-filters) are compatible with GraphQL applications as well. As with `ExecutionContext`, GraphQL apps should transform the `ArgumentsHost` object to a `GqlArgumentsHost` object.
+Nest standard [exception filters](/exception-filters) are compatible with GraphQL applications as well. You may want to implement custom response logic for them.
+
+To do this while being compatible with underlying Apollo Server error handling, we need to return an `ApolloError` instance. The instance can take up to three arguments to define the message, the error code and an extensions object that can contain any data.
 
 ```typescript
+import { Catch, ArgumentsHost, HttpException } from '@nestjs/common';
+import { GqlArgumentsHost, GqlExceptionFilter } from '@nestjs/graphql';
+import { ApolloError } from 'apollo-server-express';
```

Comment:
We should mention that this suggestion applies to Apollo driver and shouldn't be used in case someone's app uses Mercurius (or any other driver)